### PR TITLE
Point sources, adaptive source plane grid, FFT convolutions, new profiles, etc.

### PR DESCRIPTION
**Pull request by aymgal**
It's been way too long without updating the `main` branch of Herculens.

This PR solves this problem by adding several features developed over the past months:

- lensed and unlensed point source modeling, by @austinpeel 
- dynamic adaptation of the pixelated source grid, based on the mass model aand a user-provided arc mask
- FFT-based convolution of surface brightness models (from `jax.scipy`)
- Basic multi-band capabilities via a new class `LensImage3D`
- New light profile: `Multipole`, following the equivalent mass profile
- New mass profile: `PixelatedFixed`, to have a mass component entirely defined by a fixed potential/deflection field/shear/convergence (potentially on different grids)
- Sérsic light profile can now be "super-elliptical"
- Improvements in the COOLEST interface
- Improvements in plotting routines
- Bug fixes

Alongside the point source modeling, a very effcient JAX-based lens equation solver has been added (credits to @austinpeel). This will be soon moved to a separate package, such that any JAX code can import and use it easily.

### CODE REVIEW

**Comment by aymgal** [comment]
@austinpeel If you are fine with it, I will soon take care or externalizing your lens equation solver to a separate Python package (we'll see for a potential JOSS paper in a second step).
